ow_sim_tests/scripts/test_task_discard_sample.py

```diff
@@ -0,0 +1,117 @@
+    # proceed with test only when ros clock has been initialized
+    while rospy.get_time() == 0:
+      rospy.sleep(0.1)
+
+  def test_01_task_discard_sample(self):
+    task_discard_sample_result = test_arm_action(self,
+      'TaskDiscardSample', owl_msgs.msg.TaskDiscardSampleAction,
+      owl_msgs.msg.TaskDiscardSampleGoal(
+        frame = 0,
+        relative = False,
+        point = Point(2.1, 0.2, constants.DEFAULT_GROUND_HEIGHT),
+        height = 0.7
+      ),
+      TEST_NAME, "test_01_task_discard_sample",
+      server_timeout = 50.0 # (seconds) first action call needs longer timeout
+    )
+
+  def test_02_task_discard_sample(self):
+    task_discard_sample_result = test_arm_action(self,
+      'TaskDiscardSample', owl_msgs.msg.TaskDiscardSampleAction,
+      owl_msgs.msg.TaskDiscardSampleGoal(
+        frame = 0,
+        relative = False,
+        point = Point(2.1, -0.1, constants.DEFAULT_GROUND_HEIGHT),
+        height = 0.5
+      ),
+      TEST_NAME, "test_02_task_discard_sample"
+    )
+
+  def test_03_task_discard_sample(self):
+    task_discard_sample_result = test_arm_action(self,
+      'TaskDiscardSample', owl_msgs.msg.TaskDiscardSampleAction,
+      owl_msgs.msg.TaskDiscardSampleGoal(
+        frame = 0,
+        relative = False,
+        point = Point(2.0, -0.3, constants.DEFAULT_GROUND_HEIGHT),
+        height = 0.7
+      ),
+      TEST_NAME, "test_03_task_discard_sample"
+    )
+
+  def test_04_task_discard_sample(self):
+    task_discard_sample_result = test_arm_action(self,
+      'TaskDiscardSample', owl_msgs.msg.TaskDiscardSampleAction,
+      owl_msgs.msg.TaskDiscardSampleGoal(
+        frame = 0,
+        relative = True,
+        point = Point(-0.1, -0.2, 0),
+        height = 0.0
+      ),
+      TEST_NAME, "test_04_task_discard_sample"
+    )
+
+  def test_05_task_discard_sample(self):
+    task_discard_sample_result = test_arm_action(self,
+      'TaskDiscardSample', owl_msgs.msg.TaskDiscardSampleAction,
+      owl_msgs.msg.TaskDiscardSampleGoal(
+        frame = 0,
+        relative = True,
+        point = Point(-0.3, -0.15, 0.1),
+        height = 0.1
+      ),
+      TEST_NAME, "test_05_task_discard_sample"
+    )
+
+  def test_06_task_discard_sample(self):
+    task_discard_sample_result = test_arm_action(self,
+      'TaskDiscardSample', owl_msgs.msg.TaskDiscardSampleAction,
+      owl_msgs.msg.TaskDiscardSampleGoal(
+        frame = 1,
+        relative = False,
+        point = Point(0, -0.4, 0.1),
+        height = 0.0
+      ),
+      TEST_NAME, "test_06_task_discard_sample"
+    )
+
+  def test_07_task_discard_sample(self):
+    task_discard_sample_result = test_arm_action(self,
+      'TaskDiscardSample', owl_msgs.msg.TaskDiscardSampleAction,
+      owl_msgs.msg.TaskDiscardSampleGoal(
+        frame = 1,
+        relative = False,
+        point = Point(0.1, -0.1, 0.1),
+        height = 0.2
+      ),
+      TEST_NAME, "test_07_task_discard_sample"
+    )
+
```

Comment:
As I mentioned in my thread comment, lets have one more test at the end here that uses the default command values.